db/clean_planting_csv/clean_planting_API_call.py

```diff
@@ -0,0 +1,97 @@
+        central_url + "/v1/projects/1/forms/build_Tree-planting_1625058144/submissions.csv.zip",
+        headers={"Authorization": "Bearer " + email_token},
+        stream=True
+    )
+    z = zipfile.ZipFile(io.BytesIO(form_response.content))
+    z.extractall()
+
+email_token = get_email_token()
+
+if email_token:
+    get_planting_form(email_token)
+    print("raw planting API data outputted to build_Tree-planting_1625058144.csv")
+
+else:
+    print("Error getting email token")
+    
+#read in data as pandas dataframe for cleaning
+data_raw = pd.read_csv("build_Tree-planting_1625058144.csv")
+print("started cleaning and formatting raw data")
+
+#create asset_id column and fill with shortened uuid to fit in sql bigint field
+data_raw["asset_id"] = ""
+DIGITS = 16
+data_raw["asset_id"] = [int(uuid.uuid4().hex[:DIGITS], base=16) for _ in range(len(data_raw.index))]
+
+#make asset_id the first column
+first_column = data_raw.pop("asset_id")
+data_raw.insert(0, "asset_id", first_column)
+
+#split date/time column
+data_raw[["date_of_planting","time_of_planting"]] = data_raw["Time"].str.split("T", 1, expand=True)
+
+#reformat date to mm/dd/yyyy
+data_raw["date_of_planting"] = pd.to_datetime(data_raw["date_of_planting"]).dt.strftime("%m/%d/%Y")
+
+#delete unwanted columns
+data_raw = data_raw.drop(columns=["Time","Location-Accuracy", "Choose_planter","SubmissionDate", "Date", "DeviceID", "KEY"])
+data_raw = data_raw.drop(columns=["SubmitterID", "AttachmentsPresent", "AttachmentsExpected", "ReviewState", "Edits", "DeviceID.1"])
+
+#rename meta-instanceID column to clarify difference from uuid in asset_id
+data_raw = data_raw.rename(columns={"meta-instanceID": "ODK_meta-instanceID"})
+
+#complete URL in photo column, rename column to "image_url"
+#<your-server>/v1/projects/<projectId>/forms/<xmlFormId>/submissions/<instanceId>/attachments/
+data_raw["Photo_planted_sapling"] = "https://senecaparkzoo.getodk.cloud/v1/projects/1/forms/build_Tree-planting_1625058144/submissions/" + data_raw["ODK_meta-instanceID"].astype(str) + "/attachments/" + data_raw["Photo_planted_sapling"].astype(str)
+data_raw = data_raw.rename(columns={"Photo_planted_sapling":"image_url"})
+
+#clean up other column names
+data_raw = data_raw.rename(columns={"Location-Latitude":"latitude", "Location-Longitude":"longitude", "QR_Code":"tree_ID"})
+data_raw = data_raw.rename(columns={"Tree_Species":"species", "SubmitterName":"submitter_name"})
+
+#column names to lower case
+data_raw.columns= data_raw.columns.str.lower()
+
+#insert underscore for any spaces in column names
+data_raw.columns = data_raw.columns.str.replace(" ", "_")
+
+#double check that required columns are there
+if set(["asset_id","latitude","longitude"]).issubset(data_raw.columns):
+    print("passed check: all required columns in cleaned data")
+
+#output processed csv
+data_raw.to_csv("planting_processed.csv", index=False)
+print("cleaned planting data csv outputted as planting_processed.csv")
```

Comment:
This isn't necessary but I'd also look into using a common python formatted such as https://github.com/psf/black

This is purely an aesthetic thing but lots of companies usually require this sort of formatting in order to maintain consistency across code and it's good to get into the habit of using one

Reply:
Also a good tool to know about that I haven't seen before! I ran my file through Black and it re-formatted cleanly. I actually find it less readable this way, but I can see how it would be useful to conform to coding styles.